File: backend/internal/models/skill.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Skill struct {
	ID           uuid.UUID `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	UserID       uuid.UUID `gorm:"type:uuid;not null" json:"userId"`
	Name         string    `gorm:"type:varchar(100);not null" json:"name"`
	Category     string    `gorm:"type:varchar(30);not null" json:"category"`
	Level        *string   `gorm:"type:varchar(20)" json:"level"`
	IconURL      *string   `gorm:"type:text" json:"iconUrl"`
	DisplayOrder int       `gorm:"default:0" json:"displayOrder"`
	IsVisible    bool      `gorm:"default:true" json:"isVisible"`
	CreatedAt    time.Time `gorm:"not null;default:now()" json:"createdAt"`
	UpdatedAt    time.Time `gorm:"not null;default:now()" json:"updatedAt"`

	User User `gorm:"foreignKey:UserID" json:"-"`
}

File: frontend/app/types/skill.ts
export interface Skill {
  id: string
  name: string
  category?: string | null
  level?: string | null
  iconUrl?: string | null
  displayOrder: number
  isVisible: boolean
}

export interface RawSkill {
  id: string
  name: string
  category?: string | null
  level?: string | null
  icon_url?: string | null
  iconUrl?: string | null
  display_order?: number
  displayOrder?: number
  is_visible?: boolean
  isVisible?: boolean
}

export function normalizeSkill(raw: RawSkill): Skill {
  return {
    id: raw.id,
    name: raw.name,
    category: raw.category ?? null,
    level: raw.level ?? null,
    iconUrl: raw.iconUrl ?? raw.icon_url ?? null,
    displayOrder: raw.displayOrder ?? raw.display_order ?? 0,
    isVisible: raw.isVisible ?? raw.is_visible ?? true,
  }
}
